### PR TITLE
Don't track thrown non-error objects as errors

### PR DESCRIPTION
**Pull request by luismiramirez**
Remove the behaviour that accepted non-error objects in the setError()
function and added a long message to them that didn't provide useful
information.

Now any object that doesn't behave as an error will be ignored by the
setError() function. The long fallback message has also been removed.

### CODE REVIEW

**Review by unflxw — APPROVED** [review]
I think this is a much better approach than the very long message. 👏

**Review by tombruijn — APPROVED** [review]
Looks good to me for now, but what Noemi says: let's remove the isError check from other places like the React integration.

**Comment by luismiramirez** [comment]
Updated react and preact packages so they don't use `isError()` anymore

File: packages/preact/.changesets/error-type-is-no-longer-checked-at-errorboundary.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+bump: "patch"
+type: "remove"
+---
+
+Remove error type check on ErrorBoundary
```
Comment (tombruijn) [inline packages/preact/.changesets/error-type-is-no-longer-checked-at-errorboundary.md:6]:
```suggestion
Remove error type check on ErrorBoundary. This is now done in the `setError` helper instead for all function calls.
```

File: packages/react/.changesets/remove-error-type-check-on-errorboundary.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+bump: "patch"
+type: "remove"
+---
+
+Remove error type check on ErrorBoundary
```
Comment (tombruijn) [inline packages/react/.changesets/remove-error-type-check-on-errorboundary.md:6]:
```suggestion
Remove error type check on ErrorBoundary. This is now done in the `setError` helper instead for all function calls.
```

**Review by tombruijn — APPROVED** [review]

**Pull request merged**